### DISCUSSION TITLE
does the Arduino core for STM32 support C++17?

### DISCUSSION THREAD

**Discussion by PerennialNovice** [stm32duino/Arduino_Core_STM32#1895] [Q&A] [answered]
If not, what variant is it based on?
Is it possible to change it?

**Comment by fpistm** [comment]
Hi, I've never tested to build with c++17.
You can try on you own:
https://github.com/stm32duino/Arduino_Core_STM32/blob/main/platform.txt#LL64C21-L64C21

Then let us know if it is ok.

**Comment by PerennialNovice** [comment]
For a small test-project it compiles and runs without problem...

**Reply by fpistm** [reply]
Fine,
I would not move to C++17 yet because several Third party libraries are probably not compliant.

**Reply by PerennialNovice** [reply]
thanks for the warning!

**Reply by fpistm** [reply, marked as answer]
Since this #1944, C++17 is the default standard.
